internal/handlers/common/msg_setfreemonitoring.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// MsgSetFreeMonitoring is a common implementation of the setFreeMonitoring command.
+func MsgSetFreeMonitoring(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	command := document.Command()
+	var action string
+	if action, err = GetRequiredParam[string](document, "action"); err != nil {
+		return nil, err
+	}
+
+	switch action {
+	case "enable", "disable":
+		return nil, NewErrorMsg(
+			ErrFreeMonitoringDisabled,
+			"Free Monitoring has been disabled via the command-line and/or config file",
+		)
+	default:
+		return nil, NewErrorMsg(ErrBadValue, fmt.Sprintf(
+			"Enumeration value '%s' for field '%s' is not a valid value.",
+			action,
+			command+".action",
+		))
+	}
```

Comment:
@seeforschauer something like that would be good?
```suggestion
	switch action {
	case "enable", "disable":
		return nil, NewErrorMsg(
			ErrFreeMonitoringDisabled,
			"Free Monitoring has been disabled via the command-line and/or config file",
		)
	default:
		return nil, NewErrorMsg(
			ErrBadValue,
			fmt.Sprintf("Enumeration value '%s' for field '%s' is not a valid value.",
				action,
				command+".action",
			))
	}

```

Reply:
@AlekSi?
```suggestion
	switch action {
	case "enable", "disable":
		return nil, NewErrorMsg(
			ErrFreeMonitoringDisabled,
			"Free Monitoring has been disabled via the command-line and/or config file",
		)
	default:
		return nil, NewErrorMsg(
			ErrBadValue,
			fmt.Sprintf(
			        "Enumeration value '%s' for field '%s' is not a valid value.",
				action,
				command+".action",
			)
		)
	}

```